File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& str) {
        if(str.empty()) return "";
        string prefix=str[0];
        for(int i=1;i<str.size();i++){
            int j=0;
            while(j<prefix.size() && j<str[i].size() && prefix[j]==str[i][j]){
                j++;
            }
            prefix = prefix.substr(0, j);
        if (prefix == "") return ""; 
        }
        return prefix;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        first=strs[0]
        result=""
        last=strs[len(strs)-1]
        Range=min(len(first),len(last))
        for i in range(Range):
            if last[i]==first[i]:
                result+=last[i]
            else:
                break
        return result
            

        